weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/ManageColumnsButton.tsx

```diff
@@ -0,0 +1,192 @@
+      if (col.hideable ?? true) {
+        acc[col.field] = false;
+      }
+      return acc;
+    }, {} as GridColumnVisibilityModel);
+    setColumnVisibilityModel(newModel);
+  };
+  const onShowAll = () => {
+    const newModel = columnInfo.cols.reduce((acc, col) => {
+      // If a column is not hideable, we also don't need to explicitly show it.
+      if (col.hideable ?? true) {
+        acc[col.field] = true;
+      }
+      return acc;
+    }, {} as GridColumnVisibilityModel);
+    setColumnVisibilityModel(newModel);
+  };
+
+  return (
+    <>
+      <span ref={ref}>
+        <Button
+          variant="ghost"
+          icon="column"
+          tooltip="Manage columns"
+          onClick={onClick}
+        />
+      </span>
+      <Popover
+        id={id}
+        open={open}
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'center',
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'center',
+        }}
+        slotProps={{
+          paper: {
+            sx: {
+              overflow: 'visible',
+            },
+          },
+        }}
+        onClose={() => setAnchorEl(null)}
+        TransitionComponent={DraggableGrow}>
+        <Tailwind>
+          <div className="min-w-[300px] p-12">
+            <div className="handle flex items-center pb-8">
+              <div className="flex-auto font-semibold">Manage columns</div>
+              <div>
+                <Button
+                  size="small"
+                  variant="ghost"
+                  icon="close"
+                  tooltip="Close column management"
+                  onClick={e => {
+                    e.stopPropagation();
+                    setAnchorEl(null);
+                  }}
+                />
+              </div>
+            </div>
+            <div className="mb-8">
+              <TextField
+                placeholder="Filter columns"
+                autoFocus
+                value={search}
+                onChange={setSearch}
+              />
+            </div>
+            <div className="max-h-[300px] overflow-auto">
+              {columnInfo.cols.map((col: GridColDef<TraceCallSchema>) => {
+                const value = col.field;
+                const idSwitch = `toggle-vis_${value}`;
+                const checked = columnVisibilityModel[col.field] ?? true;
+                const label = col.headerName ?? value;
+                const disabled = !(col.hideable ?? true);
+                if (
+                  search &&
+                  !label.toLowerCase().includes(search.toLowerCase())
+                ) {
+                  return null;
+                }
+                return (
+                  <div key={value}>
+                    <div
+                      className={classNames(
+                        'flex items-center py-2',
+                        disabled ? 'opacity-40' : ''
```

Comment:
I noticed in [figma](https://www.figma.com/design/01KWBdMZg5QM9SRS1pQq0z/Design-System----Robot-Styles?node-id=4755-27877&t=YYZd1MLxt8BEFvrh-0) we use `opacity-30` on some of the disabled components, like checkboxes.

Reply:
not sure if applicable, just noting

Reply:
I will follow up with Yangyi.